boards: it8xxx2: enable test of tests/drivers/i2c/i2c_api

GTLin08 opened the pull request:
This enable i2c_api test on it8xxx2_evb.

GTLin08 committed:
ITE: drivers/i2c: add the API of get configuration

This commit adds the API of get_config that will make test of
tests/drivers/i2c/i2c_api pass on it8xxx2_evb board.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

keith-zephyr commented on drivers/i2c/i2c_ite_it8xxx2.c:
```diff
@@ -371,6 +376,31 @@ static int i2c_it8xxx2_configure(const struct device *dev,
+	struct i2c_it8xxx2_data *const data = dev->data;
+	uint32_t speed;
+
+	switch (data->bus_freq) {
```
Return `-EIO` if `i2c_it8xxx2_configure()` has not been called to initialize `bus_freq`.

GTLin08 replied:
Done.

GTLin08 committed:
boards: it8xxx2_evb: add aliases of i2c0

Add the aliases, we are able to build test of
tests/drivers/i2c/i2c_api for the it8xxx2_evb board.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>
